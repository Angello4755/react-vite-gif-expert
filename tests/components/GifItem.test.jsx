import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItem should', () => {

    const title = 'Angello';
    const url = 'http://localhost/url';

    test('debe hacer match con el snpshot', () => {
       const { container } = render(<GifItem title={title} url={url} />);
       expect( container ).toMatchSnapshot();
    });
    
    test('debe mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url} />);

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( title );
        
    });

    test('debe de mostrar el titulo el componente', () => {
        render(<GifItem title={title} url={url} />);
         expect( screen.getByText(title) ).toBeTruthy();
    })

})