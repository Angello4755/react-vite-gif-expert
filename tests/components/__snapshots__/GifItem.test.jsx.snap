// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should debe hacer match con el snpshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Angello"
      src="http://localhost/url"
    />
    <p>
      Angello
    </p>
  </div>
</div>
`;
